charts/prefect-worker/values.yaml

```diff
@@ -72,8 +72,35 @@ worker:
     name: null
     # -- maximum number of flow runs to start simultaneously (default: unlimited)
     limit: null
-    # -- JSON formatted base job template. If unspecified, Prefect will use the default base job template for the given worker type. If the work pool already exists, this will be ignored.
-    baseJobTemplate: null
+
+    ##  If unspecified, Prefect will use the default base job template for the given worker type. If the work pool already exists, this will be ignored.
+    ## e.g.:
+    ## baseJobTemplate:
+    ##   enabled: true
+    ##   name: "prefect-worker-base-job-template"
+    ##   data: |
+    ##    {
+    ##      "variables": {
+    ##        ...
+    ##      },
+    ##      "job_configuration": {
+    ##        ...
+    ##      }
+    ##    }
+    ## OR
+    ## baseJobTemplate:
+    ##   enabled: false
+    ##   name: "my-existing-config-map"
+    ##   data: null
+    baseJobTemplate:
+      # -- enable passing a base job template configuration to the worker via a configmap
+      enabled: false
+      # -- the name of the ConfigMap to create OR the name of an existing ConfigMap to use
+      name: prefect-worker-base-job-template
+      # -- the key to reference in the ConfigMap
+      key: baseJobTemplate.json
+      # -- JSON formatted base job template. If data is provided here, the chart will generate a configmap and mount it to the worker pod
+      data: null
```

Comment:
this totally works and im good with approving - but one alternate pattern that i've seen is:

* document the configmap's key, and expect the user to create a BYO configmap with that key in place. eg. `key` of `baseJobTemplate.json`, so you can just hardcode it in the subPath and don't have to expose it here
* rather than having an `enabled` and `data` key, i've seen bitnami chart authors just have a `configuration` key -> this would be for the current use case of defining the job template here in `values.yaml`
* rather than having a `name` key, you would make it `existingConfigMapName` -> this is for the BYO configmap case